### PR TITLE
typechecker: Disallow usage of 'continue' and 'break' outside of loops

### LINKED ISSUE #1531: `continue` and `break` accepted in random places
This should probably not compile :sweat_smile: 

```jakt
fn boog() {
    continue
}
```

Same with
```jakt
fn boog() {
	break
}
```

### PR DESCRIPTION
**Pull request by 0GreenClover0**
Fixes #1531

### CODE REVIEW

File: selfhost/typechecker.jakt [hunk 1]
```diff
@@ -6405,6 +6405,13 @@ struct Typechecker {
         Guard(expr, else_block, remaining_code, span) => .typecheck_guard(expr, else_block, remaining_code, scope_id, safety_mode, span)
     }
 
+    fn typecheck_continue(mut this, span: Span) -> CheckedStatement {
+        if not .inside_any_loop {
```
Comment (alimpfard) [inline selfhost/typechecker.jakt:6409]: This is invalid:
```rs
for foo in bar {
    defer { break }
    defer { continue }
}
```

Will have to also reject this.

Comment (alimpfard) [reply selfhost/typechecker.jakt:6409]: ...but not this:
```rs
for foo in bar { defer { for baz in quux { break } } }
```

**Comment by 0GreenClover0** [comment]
Maybe we should do all of this in parser instead, not sure 🤔

**Comment by StunxFS** [comment]
The typechecker is perfect for detecting invalid usages

**Pull request merged**